File: Solutions/876. Middle of the Linked List/876. Middle of the Linked List.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func middleNode(head *ListNode) *ListNode {
    slow,fast:=head,head;
    for fast!=nil && fast.Next!=nil{
        fast=fast.Next.Next
        slow=slow.Next
    }
    return slow
}

func middleNode(head *ListNode) *ListNode {
    var A []*ListNode
    A=append(A,head)
    for A[len(A)-1].Next!=nil{
        A=append(A,A[len(A)-1].Next)
    }
    return A[len(A)/2]
}

File: Solutions/876. Middle of the Linked List/876. Middle of the Linked List.java
/**
 * 
 */
package github.com.ErnestL1n;

/**
 * @author https://github.com/ErnestL1n
 *
 */
class ListNode{
	int val;
	ListNode next;
	ListNode(){};
	ListNode(int value){this.val=value;};
	ListNode(ListNode next,int value){
		this.next=next;
		this.val=value;
	};
	
}
public class Middle_of_the_Linked_List {

	/**
	 * @param args
	 */
	
    public ListNode middleNodeFastAndSlow(ListNode head) {
        ListNode slow=head,fast=head;
        while(fast!=null&&fast.next!=null){
            slow=slow.next;
            fast=fast.next.next;
        }
        return slow;
    }
    
    public ListNode middleNode(ListNode head) {
        ListNode[] A=new ListNode[100];
        int t=0;
        while(head.next!=null){
            A[t++]=head;
            head=head.next;
        }
        return A[t/2];
    }
	public static void main(String[] args) {

	}

}
